File: src/alerts/implementations/file_renamed.rs
use crate::{
    alerts::FileRenamedAlert,
    ffi::alerts::file_renamed::ffi::{
        file_renamed_alert_get_new_name, file_renamed_alert_get_old_name,
    },
    torrent_handle::TorrentHandle,
};

impl FileRenamedAlert {
    #[inline(always)]
    pub fn handle(&self) -> TorrentHandle {
        self.as_torrent_alert().handle()
    }

    #[inline(always)]
    pub fn torrent_name<'a>(&'a self) -> &'a str {
        self.as_torrent_alert().torrent_name()
    }

    #[inline(always)]
    pub fn message(&self) -> String {
        self.as_torrent_alert().message()
    }

    /// The file index within the torrent that was renamed.
    #[inline(always)]
    pub fn index(&self) -> i32 {
        unsafe {
            crate::ffi::alerts::file_renamed::ffi::file_renamed_alert_get_index(self.0)
        }
    }

    pub fn old_name<'a>(&'a self) -> &'a str {
        unsafe { file_renamed_alert_get_old_name(self.0) }
    }

    pub fn new_name<'a>(&'a self) -> &'a str {
        unsafe { file_renamed_alert_get_new_name(self.0) }
    }
}

File: cpp/alerts/file_renamed.cpp
#include "./file_renamed.h"


namespace ltrs {
    rust::Str file_renamed_alert_get_old_name(lt::file_renamed_alert* alert) {
        return alert->old_name();
    }

    rust::Str file_renamed_alert_get_new_name(lt::file_renamed_alert* alert) {
        return alert->new_name();
    }

    int32_t file_renamed_alert_get_index(lt::file_renamed_alert* alert) {
        return static_cast<int32_t>(alert->index);
    }
}
